src/EVMBlackjack.sol

```diff
@@ -0,0 +1,337 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.16;
+
+import {Chip} from "../src/Chip.sol";
+
+contract EVMBlackjack {
+    /*//////////////////////////////////////////////////////////////
+    //  Events
+    //////////////////////////////////////////////////////////////*/
+
+    event BetPlaced(address indexed player, uint16 betSize);
+    event PlayerCardDealt(address indexed player, uint8 card, uint8 handIndex);
+    event DealerCardDealt(address indexed player, uint8 card);
+    event InsuranceTaken(address indexed player, bool take);
+    event PlayerActionTaken(address indexed player, Action action);
+    event PlayerBust(address indexed player);
+    event DealerBust(address indexed player);
+    event PayoutsHandled(address indexed player, uint16 playerPayout, uint16 dealerPayout);
+
+    /*//////////////////////////////////////////////////////////////
+    //  Errors
+    //////////////////////////////////////////////////////////////*/
+
+    error InvalidStateTransition();
```

Comment:
this error will only be thrown in response to an invalid action, right? could we retitle as InvalidAction to make it slightly less opaque to the caller?

Reply:
Resolved